Sources/Crashes/CrashReportSender.swift

```diff
@@ -18,29 +18,87 @@
 
 import Foundation
 import MetricKit
+import Common
 
-public final class CrashReportSender {
+public protocol CrashReportSending {
+    var pixelEvents: EventMapping<CrashReportSenderError>? { get }
 
+    init(platform: CrashCollectionPlatform, pixelEvents: EventMapping<CrashReportSenderError>?)
+
+    func send(_ crashReportData: Data, crcid: String?) async -> (result: Result<Data?, Error>, response: HTTPURLResponse?)
+    func send(_ crashReportData: Data, crcid: String?, completion: @escaping (_ result: Result<Data?, Error>, _ response: HTTPURLResponse?) -> Void)
+}
+
+public enum CrashReportSenderError: Error {
+    case crcidMissing
+    case submissionFailed(HTTPURLResponse?)
+}
+
+public final class CrashReportSender: CrashReportSending {
     static let reportServiceUrl = URL(string: "https://duckduckgo.com/crash.js")!
+
+    static let httpHeaderCRCID = "crcid"
+
     public let platform: CrashCollectionPlatform
+    public var pixelEvents: EventMapping<CrashReportSenderError>?
 
-    public init(platform: CrashCollectionPlatform) {
+    private let session = URLSession(configuration: .ephemeral)
+
+    public init(platform: CrashCollectionPlatform, pixelEvents: EventMapping<CrashReportSenderError>?) {
         self.platform = platform
+        self.pixelEvents = pixelEvents
     }
 
-    public func send(_ crashReportData: Data) async {
+    public func send(_ crashReportData: Data, crcid: String?, completion: @escaping (_ result: Result<Data?, Error>, _ response: HTTPURLResponse?) -> Void) {
```

Comment:
The `crcid` argument here is optional, but the storage is returning a non-nil `String`, as it defaults to returning an empty string. I'd prefer to update the storage API (`public var crcid: String`) to return `String?` and avoid defaulting to an empty string, since it gives us type-safety around ensuring we have a _real_ CRC ID.